### PR TITLE
Gait mirroring

### LINKED ISSUE #19: Import gait only in main screen
When importing a gait, a new graph for each joint is opened. This is not useful, so the imported gait has to be opened only in the main screen.

### PR DESCRIPTION
**Pull request by Ishadijcks**
Add gait mirroring, when it's enabled the current gait is saved as 2 gaits. The user enters 2 keys, for example 'left' and 'right'. 2 subgaits are created, one with the normal name and one with key 1 replaced with key 2 (or the other way around). Then for the name of each joint, the key is swapped as well.

## Features
- Add custom layout, in the `config/layout.yaml` you can determine which joint editor should go where.
- Fix a bug where the jointsettings would load twice (might fix #19)
- Allow custom keys to be set from the ui

### CODE REVIEW

File: march_rqt_gait_generator/src/march_rqt_gait_generator/model/Gait.py [hunk 1]
```diff
@@ -112,3 +115,28 @@ def set_duration(self, duration, rescale=False):
...
+            elif key_2 in joint.name:
+                mirrored_name = joint.name.replace(key_2, key_1)
+            else:
+                continue
+
+            mirrored_joint = Joint(mirrored_name, joint.limits, joint.setpoints, joint.duration)
+            mirrored_joints.append(mirrored_joint)
+
+        return Gait(mirrored_joints, self.duration, self.name, mirrored_subgait_name, self.version, self.description)
+
```
Comment (Ishadijcks) [inline march_rqt_gait_generator/src/march_rqt_gait_generator/model/Gait.py]:
```suggestion
```

**Comment by Ishadijcks** [comment]
I have added better error handling to the mirror functionality.

The problem you encounter occurs when the last setpoint is not at the duration for some joints I believe. Can you open another issue for this?

**Comment by bentebloks** [comment]
Still gives an error when the name of the subgait does not contain 'right' or 'left'. However, the gait for one leg is saved successfully. Also this message pops up on the screen: 'Could not mirror subgait. Subgait does not require key left.' After this, the program closes itself.

This error is produced:
```
march@march-OptiPlex-980:~/march_ws/src/gait-generation$ roslaunch march_rqt_gait_generator march_rqt_gait_generator.launch 
... logging to /home/march/.ros/log/1d74e67c-8851-11e9-8ee2-a4badbf06f2e/roslaunch-march-OptiPlex-980-30384.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://march-OptiPlex-980:37941/

SUMMARY
========

PARAMETERS
 * /joint_layout/left_ankle/x: 2
 * /joint_layout/left_ankle/y: 0
 * /joint_layout/left_hip/x: 0
 * /joint_layout/left_hip/y: 0
 * /joint_layout/left_knee/x: 1
 * /joint_layout/left_knee/y: 0
 * /joint_layout/right_ankle/x: 2
 * /joint_layout/right_ankle/y: 1
 * /joint_layout/right_hip/x: 0
 * /joint_layout/right_hip/y: 1
 * /joint_layout/right_knee/x: 1
 * /joint_layout/right_knee/y: 1
 * /robot_description: <?xml version="1....
 * /rosdistro: kinetic
 * /rosversion: 1.12.14

NODES
  /
    march_rqt_gait_generator (march_rqt_gait_generator/rqt_gait_generator)
    robot_state_publisher (robot_state_publisher/robot_state_publisher)

auto-starting new master
process[master]: started with pid [30397]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 1d74e67c-8851-11e9-8ee2-a4badbf06f2e
process[rosout-1]: started with pid [30410]
started core service [/rosout]
process[robot_state_publisher-2]: started with pid [30427]
process[march_rqt_gait_generator-3]: started with pid [30428]
Unknown tag: hardwareInterface
Unknown tag: hardwareInterface
Unknown tag: hardwareInterface
Unknown tag: hardwareInterface
Unknown tag: hardwareInterface
Unknown tag: hardwareInterface
Unknown tag: plugin
[INFO] [1559821759.975871]: Skipping fixed joint hip
[ INFO] [1559821760.382938369]: Stereo is NOT SUPPORTED
[ INFO] [1559821760.382995141]: OpenGl version: 3 (GLSL 1.3).
appmenu-qt: handleReparent 143 The given QMenuBar is already registered by appmenu-qt5, skipping
load gait into ui
[INFO] [1559821761.168711]: Restor settings called here: 3
[INFO] [1559821826.809641]: Selected output directory /home/march/Documents/Gaits
[INFO] [1559821828.245017]: Selected output directory /home/march/Documents/Gaits
[INFO] [1559821828.246888]: Writing gait to /home/march/Documents/Gaits/Test/Open/First_try.subgait
[INFO] [1559821828.295596]: Selected output directory /home/march/Documents/Gaits
Traceback (most recent call last):
  File "/home/march/march_ws/src/gait-generation/march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py", line 78, in <lambda>
    export_to_file(self.gait.get_mirrored("left", "right"), self.get_gait_directory()),
  File "/home/march/march_ws/src/gait-generation/march_rqt_gait_generator/src/march_rqt_gait_generator/import_export.py", line 20, in export_to_file
    subgait.trajectory = gait.to_joint_trajectory()
AttributeError: 'bool' object has no attribute 'to_joint_trajectory'
[march_rqt_gait_generator-3] process has died [pid 30428, exit code 1, cmd /home/march/march_ws/src/gait-generation/march_rqt_gait_generator/scripts/rqt_gait_generator __name:=march_rqt_gait_generator __log:=/home/march/.ros/log/1d74e67c-8851-11e9-8ee2-a4badbf06f2e/march_rqt_gait_generator-3.log].
log file: /home/march/.ros/log/1d74e67c-8851-11e9-8ee2-a4badbf06f2e/march_rqt_gait_generator-3*.log
^C[robot_state_publisher-2] killing on exit
[rosout-1] killing on exit
[master] killing on exit
shutting down processing monitor...
... shutting down processing monitor complete
done

```

**Comment by Ishadijcks** [comment]
Fixed, it was acting a bit weird when joints with the key names didn't exist.

**Pull request merged**